Rewrite discussion (#47)
greenelab/text_mined_hetnet_manuscript: content/05.discussion.md

## Discussion

We measured the extent to which label functions can be re-used across multiple edge types to extract relationships from literature.
Through our sampling experiment, we found that adding edge-specific label functions increases performance for the generative model ({@fig:auroc_gen_model_performance}.
We found that label functions designed from relatively related edge types can increase performance (Gene interacts Gene (GiG) label functions predicting the Compound binds Gene (CbG) edge and vice versa), while the Disease associates Gene (DaG) edge type remained agnostic to label function sources (Figure {@fig:auroc_gen_model_performance} and Supplemental Figure {@fig:aupr_gen_model_performance}).
Furthermore, we found that using all label functions at once generally hurts performance with the exception being the DaG edge type (Figure {@fig:auroc_grabbag_gen_model_test_set} and Supplemental Figure {@fig:aupr_grabbag_gen_model_test_set}).

> We premise that DaG is a broadly defined edge type, e.g. can contain sentences from other edge types such as Disease (up/down)regulating a gene, which makes it agnostic to label function sources (examples highlighted in our [annotated sentences](https://github.com/greenelab/text_mined_hetnet_manuscript/tree/master/supplementary_materials/annotated_sentences)).  

Reviewer: ```suggestion
One possibility for this observation is that DaG is a broadly defined edge type, e.g., "associates" contains many concepts related to other edge types such as Disease (up/down)regulating a gene, which makes it more agnostic to label function sources (examples highlighted in our [annotated sentences](https://github.com/greenelab/text_mined_hetnet_manuscript/tree/master/supplementary_materials/annotated_sentences)).  
```

> Regarding the discriminator model, adding edge-specific label function immensely improved performance for two out of the four edge types (Compound treats Disease (CtD) and DaG) ({@fig:auroc_disc_model_performance}). 

Reviewer: ```suggestion
Regarding the discriminator model, adding edge-specific label function substantially improved performance for two out of the four edge types (Compound treats Disease (CtD) and DaG) ({@fig:auroc_disc_model_performance}). 
```

GiG and CbG discriminator models showed minor improvements compared to the generative model, but only when nearly all edge-specific label functions are included.

> Potential explanation for this performance is the negative class bias both edge types had in their datasets (Table {@tbl:candidate-sentences}).

Reviewer: I don't understand this. I'd either remove it or add a bit more detail around it so that a reader can get the reasoning here.

> Plus, the GiG dataset contained a prodigious amount of spurious gene mentions, because Pubtator [@doi:10.1093/nar/gkt441] tags gene abbreviations along with gene symbols.

Reviewer: Is there evidence for this in the paper, or is this something you anecdotally noticed and want to mention? Either is fine, but if it's just something you noticed while doing this then we should couch it a bit more cautiously.

Author: It is the latter. I noticed this observation while working on this project.

Also, we encountered difficulty in calibrating each discriminator model (Figure {@fig:discriminative_model_calibration}).

> The temperature scaling algorithm was able to calibrate the high scores for each model, but could not fully calibrate each model. 

Reviewer: ```suggestion
The temperature scaling algorithm appears to improve calibration for the highest scores for each model but did not successfully calibrate throughout the entire range of predictions. 
```

> A reason for the poor calibration is that our labeled datasets are too small (Table {@tbl:candidate-sentences}) and calibration could improve with more examples.

Reviewer: ```suggestion
Improving performance throughout the range may require more labeled examples or may be a limitation of the approach in this setting.
```

> Lastly, we recalled a great number of existing edges in Hetionet v1 as well as detect over thousands of new edges to each edge type (Figure {@fig:hetionet_reconstruction}).

Reviewer: ```suggestion
Even with these limitations, this early-stage approach could recall many existing edges from an existing knowledge base, Hetionet v1, and suggest many new high-confidence edges for inclusion.
```

> Our findings highlight the effectiveness of this framework and show that this framework can establish new edges within Hetionet v1.  

Reviewer: ```suggestion
Our findings suggest that further work, including an expansion of edge types and a move to full text from abstracts, may make this approach suitable for building continuously updated knowledge bases to address drug repositioning and other biomedical challenges.  
```
